### PR TITLE
Show output during batch deprovision

### PR DESCRIPTION
**Pull request by jorissteyn**
Example output, only shown without --json-only:

    [  0% of 3 users]    Working on urn:collab:person:example.com:admin...
    [ 33% of 3 users]    Working on urn:collab:person:example.com:admin2...
    [ 66% of 3 users]    Working on urn:collab:person:example.com:admin3...
    [100% of 3 users]    Done.

See https://www.pivotaltracker.com/story/show/158107568

### CODE REVIEW

**Review by MKodde — COMMENTED** [review]
Adding an extra whitespace between the last progress output and the `15 deprovision calls to services failed. See error messages below:` would visually be more pleasing.

Otherwise a very nice addition!

```bash
$ ./userlifecycle deprovision --dry-run
Continue with deprovisioning? (y/n) y
[  0% of 3 users]    Working on collab:person:user:example.com:admin2...
[ 33% of 3 users]    Working on collab:person:user:example.com:jc...
[ 66% of 3 users]    Working on collab:person:user:example.com:lemmy...
[100% of 3 users]    Done.
15 deprovision calls to services failed. See error messages below:

 * engine.example.org: "Resource "/deprovision/collab%3Aperson%3Auser%3Aexample.com%3Aadmin2" not found" for user "collab:person:user:example.com:admin2"
 * engine2.example.org: "Resource "/deprovision/collab%3Aperson%3Auser%3Aexample.com%3Aadmin2" not found" for user "collab:person:user:example.com:admin2"
```

**Comment by jorissteyn** [comment]
The progress report now always ends with a newline.

File: src/OpenConext/UserLifecycle/Domain/Service/ProgressReporterInterface.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\UserLifecycle\Domain\Service;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+interface ProgressReporterInterface
```
Comment (MKodde) [inline src/OpenConext/UserLifecycle/Domain/Service/ProgressReporterInterface.php]: Not 100% sure this should be in the domain layer. What is your motivation for adding it here? 

My rationale is: reporting on the progress of the batch deprovision command is not really important to the domain and fits better in the Application layer.

File: tests/integration/UserLifecycleBundle/Command/BatchDeprovisionCommandTest.php [hunk 2]
```diff
@@ -97,7 +98,9 @@ public function setUp()
         $logger = m::mock(LoggerInterface::class);
         $logger->shouldIgnoreMissing();
 
-        $this->application->add(new DeprovisionCommand($deprovisionService, $summaryService, $logger));
+        $progressReporter = new ProgressReporter();
+
+        $this->application->add(new DeprovisionCommand($deprovisionService, $summaryService, $progressReporter, $logger));
```
Comment (MKodde) [inline tests/integration/UserLifecycleBundle/Command/BatchDeprovisionCommandTest.php:103]: Have you considered adding some assertions to the tests, ascertaining the correct progress is outputted? 

On a sidenote, I myself am guilty of not fully testing the output in the integration tests.

**Comment by MKodde** [comment]
I started with a functional review, I've added two code review notes.

File: src/OpenConext/UserLifecycle/Domain/Service/ProgressReporterInterface.php (hunk 1 above)
Comment (jorissteyn) [reply src/OpenConext/UserLifecycle/Domain/Service/ProgressReporterInterface.php]: We've now created a new interface for the console-specific parts.

File: tests/integration/UserLifecycleBundle/Command/BatchDeprovisionCommandTest.php (hunk 2 above)
Comment (jorissteyn) [reply tests/integration/UserLifecycleBundle/Command/BatchDeprovisionCommandTest.php:103]: The reporter has its own test. Choosing not to test the specific messages / numbers the services uses to report the progress.

**Pull request merged**